pageserver/src/tenant/timeline.rs

```diff
@@ -4756,6 +4770,18 @@ impl Timeline {
     }
 }
 
+impl Drop for Timeline {
+    fn drop(&mut self) {
+        if let Some(ancestor) = &self.ancestor_timeline {
+            ancestor
+                .gc_info
+                .write()
+                .unwrap()
```

Comment:
panic in destructor will directly abort the process, better to log/ignore the error here instead of unwrap.

Reply:
or we should do this in `remove_timeline_from_tenant` function?

Reply:
Good catch, amended in df659f2098ff2fc3a7d347794d9c247001e5f799.

I think the neatest thing is to keep it in the destructor to preserve the mental model of the Timeline being its own RAII guard for parent state.